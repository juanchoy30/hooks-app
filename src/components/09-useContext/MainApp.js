import React from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

export const MainApp = () => {

    const user = {
        id: 1234,
        name: 'Juan',
        email: 'juanchoy30@gmail.com'
    }

    return (
        <UserContext.Provider value={ user }>

            <AppRouter />

        </UserContext.Provider>
    )
}
