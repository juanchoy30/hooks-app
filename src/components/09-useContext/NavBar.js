import React from 'react';
import { Link, NavLink } from 'react-router-dom'; //NavLink and Link are the same, but NavLink can incorporate independent CSS styles, when active

export const NavBar = () => {
    return (

        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">


            <NavLink className="navbar-brand" to="/"> useContext </NavLink>

            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <NavLink exact activeClassName="active" to="/" className="nav-link"> Home </NavLink>
                    <NavLink exact activeClassName="active" to="/about"  className="nav-link"> About </NavLink>
                    <NavLink exact activeClassName="active" to="/login"  className="nav-link"> Login </NavLink>
                </div>
            </div>
        </nav>



    )
}
