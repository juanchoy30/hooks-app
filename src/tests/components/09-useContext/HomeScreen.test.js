import React from 'react';
import { mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Tests on <HomeScreen />', () => {

    const user = {
        name: 'Juan',
        email: 'juanchoy30@gmail.com'
    }

    const wrapper = mount( 
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen /> 
        </UserContext.Provider>
    );
    
    test('should render correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });
    


})
