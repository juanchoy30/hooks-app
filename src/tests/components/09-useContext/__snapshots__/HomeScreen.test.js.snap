// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <HomeScreen /> should render correctly 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Juan",
   "email": "juanchoy30@gmail.com"
}
  </pre>
</div>
`;
